zerver/lib/outgoing_webhook.py

```diff
@@ -8,17 +8,45 @@
 import logging
 from six.moves import urllib
 from functools import reduce
+from requests import Response
 
 from django.utils.translation import ugettext as _
 
-from zerver.models import Realm, get_realm_by_email_domain, get_user_profile_by_id, get_client
+from zerver.models import Realm, UserProfile, get_realm_by_email_domain, get_user_profile_by_id, get_client
 from zerver.lib.actions import check_send_message
 from zerver.lib.queue import queue_json_publish
 from zerver.lib.validator import check_dict, check_string
 from zerver.decorator import JsonableError
 
 MAX_REQUEST_RETRIES = 3
 
+class OutgoingWebhookServiceInterface(object):
+
+    def __init__(self, base_url, token, user_profile, service_name):
+        # type: (Text, Text, UserProfile, Text) -> None
+        self.base_url = None # type: Text
+        self.token = None # type: Text
+        self.user_profile = None  # type: Text
+        self.service_name = None  # type: Text
+
+    # would receive the trigger data and return rest operation related info
```

Comment:
When implementing an API like this, I think it's very important to write very clear comments about each method.  This is a great start, but I would still have a few open questions if I were looking at this:

- Where does the "trigger data" come from?  (Does it originate from a Zulip message?  Does the client have to create one by hand?)  What formats should be expected?
- What does "rest operation related info and data" mean?  What are the requirements for the structure of the return values?

Ideally, someone would be able to answer these questions just by looking at the comment (without needing to look at any other code).